### PR TITLE
[WIP] Use complete header version

### PR DESCRIPTION
**Pull request by mikew-lunarg**
This must wait for the next header update before CI and merge.

### CODE REVIEW

File: build-gn/generate_vulkan_layers_json.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def main():
             for line in infile:
                 line = line.replace('@RELATIVE_LAYER_BINARY@',
                                     relative_path_prefix + layer_lib_name)
-                line = line.replace('@VK_VERSION@', '1.1.' + vk_version)
```
Comment (mikes-lunarg) [inline build-gn/generate_vulkan_layers_json.py]: Whoops, this should be 1.2

File: build-gn/generate_vulkan_layers_json.py [hunk 2]
```diff
@@ -87,7 +87,7 @@ def main():
     vk_version = None
     with open(vk_header_filename) as vk_header_file:
         for line in vk_header_file:
-            if line.startswith('#define VK_HEADER_VERSION'):
+            if line.startswith('#define VK_HEADER_VERSION_COMPLETE'):
```
Comment (mikes-lunarg) [inline build-gn/generate_vulkan_layers_json.py]: This breaks the json generation in the GN build, output ends up looking like this:
```
...
    "layer" : {
        "name": "VK_LAYER_KHRONOS_validation",
        "type": "GLOBAL",
        "library_path": "../libVkLayer_khronos_validation.so",
        "api_version": "VK_HEADER_VERSION)",
...
```
It should be possible construct a string version of `VK_HEADER_VERSION_COMPLETE` but would require a bit more regex hackery.  CMake is still hardcoding the "1.2." prefix for layer json versioning so we probably want to keep the original approach here as well.

**Comment by mikew-lunarg** [comment]
yeah, I should tag this as WIP, it's premature for review as you've noticed.

**Comment by mark-lunarg** [comment]
Is there an update on this PR, or can we close it for now?

**Comment by mikew-lunarg** [comment]
Please keep this alive for new.   I have this on my list, and will finish or close soon.

**Pull request closed without merging**